comp/logs/agent/config/integration_config.go

```diff
@@ -35,59 +35,59 @@ const (
+	TailingMode  string   `mapstructure:"start_position" json:"start_position,omitempty"` // File
 
 	//nolint:revive // TODO(AML) Fix revive linter
-	ConfigId           string   `mapstructure:"config_id" json:"config_id"`                   // Journald
-	IncludeSystemUnits []string `mapstructure:"include_units" json:"include_units"`           // Journald
-	ExcludeSystemUnits []string `mapstructure:"exclude_units" json:"exclude_units"`           // Journald
-	IncludeUserUnits   []string `mapstructure:"include_user_units" json:"include_user_units"` // Journald
-	ExcludeUserUnits   []string `mapstructure:"exclude_user_units" json:"exclude_user_units"` // Journald
-	IncludeMatches     []string `mapstructure:"include_matches" json:"include_matches"`       // Journald
-	ExcludeMatches     []string `mapstructure:"exclude_matches" json:"exclude_matches"`       // Journald
-	ContainerMode      bool     `mapstructure:"container_mode" json:"container_mode"`         // Journald
+	ConfigId           string   `mapstructure:"config_id" json:"config_id,omitempty"`                   // Journald
+	IncludeSystemUnits []string `mapstructure:"include_units" json:"include_units,omitempty"`           // Journald
+	ExcludeSystemUnits []string `mapstructure:"exclude_units" json:"exclude_units,omitempty"`           // Journald
+	IncludeUserUnits   []string `mapstructure:"include_user_units" json:"include_user_units,omitempty"` // Journald
+	ExcludeUserUnits   []string `mapstructure:"exclude_user_units" json:"exclude_user_units,omitempty"` // Journald
+	IncludeMatches     []string `mapstructure:"include_matches" json:"include_matches,omitempty"`       // Journald
+	ExcludeMatches     []string `mapstructure:"exclude_matches" json:"exclude_matches,omitempty"`       // Journald
+	ContainerMode      bool     `mapstructure:"container_mode" json:"container_mode,omitempty"`         // Journald
 
-	Image string // Docker
-	Label string // Docker
+	Image string `json:"image,omitempty"` // Docker
+	Label string `json:"label,omitempty"` // Docker
 	// Name contains the container name
-	Name string // Docker
+	Name string `json:"name,omitempty"` // Docker
 	// Identifier contains the container ID.  This is also set for File sources and used to
 	// determine the appropriate tags for the logs.
-	Identifier string // Docker, File
+	Identifier string `json:"identifier,omitempty"` // Docker, File
 
-	ChannelPath string `mapstructure:"channel_path" json:"channel_path"` // Windows Event
-	Query       string // Windows Event
+	ChannelPath string `mapstructure:"channel_path" json:"channel_path,omitempty"` // Windows Event
+	Query       string `json:"query,omitempty"`                                    // Windows Event
 
 	// used as input only by the Channel tailer.
 	// could have been unidirectional but the tailer could not close it in this case.
-	Channel chan *ChannelMessage
+	Channel chan *ChannelMessage `json:"-"`
```

Comment:
Do you know what the JSON parser will do with a key of `"-"` (since now multiple fields have that key)? 
This struct is also populated by parsing configs (JSON and YAML). While it's unlikely, I want to avoid any possible breaking changes.

Reply:
I did a manual `PublicJSON` export so we're sure not to break anything + addressing your second comment, where we explicitly say the fields we send back to the backend